modules/collector-source/pkg/collector/diagnostics.go

```diff
@@ -0,0 +1,78 @@
+package collector
+
+import (
+
+	"github.com/opencost/opencost/core/pkg/source"
+	"github.com/opencost/opencost/modules/collector-source/pkg/metric"
+)
+
+// Collector Metric Diagnostic IDs
+const (
+	// KubecostDiagnosticMetricID is the identifier for the metric used to determine if Kubecost metrics are being scraped.
+	KubecostDiagnosticMetricID = "kubecostMetric"
+
+	// NodesDiagnosticMetricID is the identifier for the query used to determine if the node CPU cores capacity is being scraped
+	NodesDiagnosticMetricID = "nodesCPUMetrics"
+)
+
+
+// diagnostic definitions mapping holds all of the diagnostic definitions that can be used for collector metrics diagnostics
+var diagnosticDefinitions map[string]*diagnosticDefinition = map[string]*diagnosticDefinition{
+	NodesDiagnosticMetricID: {
+		ID:          NodesDiagnosticMetricID,
+		MetricsId:   metric.NodeCPUCoresCapacityID,
+		Label:       "Node CPU cores capacity is being scraped",
+		Description: "Determine if the node CPU cores capacity is being scraped",
+		QueryFmt:    "avg(avg_over_time(kube_node_status_capacity_cpu_cores{}[1m])) by (cluster_id, node)",
+	},
+
+	KubecostDiagnosticMetricID: {
+		ID:          KubecostDiagnosticMetricID,
+		MetricsId:   metric.NodeActiveMinutesID,
+		Label:       "Kubecost metrics for a node are being scraped",
+		Description: "Determine if kubecost metrics for a node are being scraped",
+		QueryFmt:    "avg(node_total_hourly_cost{}[1m]) by (node, cluster_id, provider_id)",
+	},
+}
+
+type diagnosticDefinition struct {
+	ID          string
+	MetricsId   metric.MetricCollectorID
+	QueryFmt    string
```

Comment:
QueryFmt doesn't really make sense in this context. you could print out the type of the aggregator being used and the metric name that is aggregated.